spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/SparkBinPackPositionDeletesRewriter.java

```diff
@@ -64,6 +64,8 @@ class SparkBinPackPositionDeletesRewriter extends SizeBasedPositionDeletesRewrit
     // Disable Adaptive Query Execution as this may change the output partitioning of our write
     this.spark = spark.cloneSession();
     this.spark.conf().set(SQLConf.ADAPTIVE_EXECUTION_ENABLED().key(), false);
+    // Because we're turning off AQE we need to turn off shuffle partitions b/c deletes rewriter makes a join later
+    this.spark.conf().set(SQLConf.SHUFFLE_PARTITIONS().key(), "1");
```

Comment:
can you please add some tests to `TestRewritePositionDeleteFilesProcedure` that reproduce the behavior you were seeing in #9833?

Reply:
I checked and the test you added passes for me even when this isn't being forced to 1